python/dgl/distributed/sparse_emb.py

```diff
@@ -13,31 +13,32 @@ class SparseNodeEmbedding:
     ----------
     g : DistGraph
         The distributed graph object.
-    name : str
-        The name of the embeddings
     shape : tuple of int
         The shape of the embedding. The first dimension should be the number of nodes.
-    initializer : callable
+    name : str
+        The name of the embeddings
+    init_func : callable
         The function to create the initial data.
+    part_policy : PartitionPolicy
+        The partition policy.
 
     Examples
     --------
     >>> emb_init = lambda shape, dtype: F.zeros(shape, dtype, F.cpu())
     >>> shape = (g.number_of_nodes(), 1)
-    >>> emb = dgl.distributed.SparseNodeEmbedding(g, 'emb1', shape, emb_init)
+    >>> emb = dgl.distributed.Embedding(g, 'emb1', shape, emb_init)
     >>> optimizer = dgl.distributed.SparseAdagrad([emb], lr=0.001)
     >>> for blocks in dataloader:
     >>>     feats = emb(nids)
     >>>     loss = F.sum(feats + 1, 0)
     >>>     loss.backward()
     >>>     optimizer.step()
     '''
-    def __init__(self, g, name, shape, initializer):
-        assert shape[0] == g.number_of_nodes()
-        part_policy = PartitionPolicy(NODE_PART_POLICY, g.get_partition_book())
-        g.ndata[name] = DistTensor(g, shape, F.float32, name, part_policy, initializer)
+    def __init__(self, g, shape, name, init_func, part_policy=None):
+        if part_policy is None:
+            part_policy = PartitionPolicy(NODE_PART_POLICY, g.get_partition_book())
 
-        self._tensor = g.ndata[name]
+        self._tensor = DistTensor(g, shape, F.float32, name, init_func, part_policy)
         self._trace = []
 
     def __call__(self, idx):
```

Comment:
How to delete the old gradient stored in trace?

Reply:
the trace is deleted in optimizer. but there is a bug if the optimizer isn't called, for example, in the inference. this is something we need to fix.